### PR TITLE
Docs updates

### PR DESCRIPTION
**Pull request by FGasper**
This clarifies why to avoid encode_utf8() and decode_utf8() and expands
on the problem of decoding “utf8”.

### CODE REVIEW

File: Encode.pm [hunk 1]
```diff
@@ -927,6 +929,20 @@ important distinction between C<"UTF-8"> and C<"utf8">.
...
+
+  $s = decode("utf8", "\xf8\x88\x80\x80\x80");
+
+C<"UTF-8">, by contrast, will either coerce the input to something valid:
+
+    $s = decode("UTF-8", "\xf8\x88\x80\x80\x80"); # U+FFFD
+
+.. or croak:
+
+    decode("UTF-8", "\xf8\x88\x80\x80\x80", 9);
```
Comment (pali) [inline Encode.pm:944]: Hello! Can you put human readable constant names as a third _check_ argument instead of magic number 9?

File: Encode.pm [hunk 2]
```diff
@@ -499,30 +499,32 @@ followed by C<encode> as follows:
 
   $octets = encode_utf8($string);
 
+B<WARNING>: L</UTF-8 vs. utf8 vs. UTF8|This function can produce invalid UTF-8!>
```
Comment (pali) [inline Encode.pm:502]: Have you checked output of your change?

I think that this hyperlink does not work.

POD syntax for hyperlinks is `L<text|name/section>` which looks like you swapped _text_ and _link_ parts.

**Review by pali — CHANGES_REQUESTED** [review]

File: Encode.pm (hunk 1 above)
Comment (FGasper) [reply Encode.pm:944]: I was shooting for consistency with the other examples above, which give 1 instead of FB_CROAK. Should I update those as well?

Comment (dankogai) [reply Encode.pm:944]: I've overlooked this.  `9` must be ` DIE_ON_ERR|LEAVE_SRC`.  I'll fix it.

File: Encode.pm (hunk 2 above)
Comment (dankogai) [reply Encode.pm:502]: I'll fix this one as well

File: Encode.pm [hunk 3]
```diff
@@ -499,30 +499,32 @@ followed by C<encode> as follows:
...
 
-B<WARNING>: do not use this function for data exchange as it can produce
-not strict utf8 $octets! For strictly valid UTF-8 output use
-C<$octets = encode("UTF-8", $string)>.
-
 =head3 decode_utf8
 
   $string = decode_utf8($octets [, CHECK]);
 
+B<WARNING>: L</UTF-8 vs. utf8 vs. UTF8|This function accepts invalid UTF-8!>
```
Comment (pali) [inline Encode.pm:516]: Also need to fix this hyperlink.

File: Encode.pm (hunk 1 above)
Comment (pali) [reply Encode.pm:944]: Maybe `FB_CROAK` instead of `DIE_ON_ERR` for consistency?

Comment (dankogai) [reply Encode.pm:944]: fixed in https://github.com/dankogai/p5-encode/commit/67f59dd4092c12043182e427dfaa5abd5c791e58

File: Encode.pm (hunk 3 above)
Comment (dankogai) [reply Encode.pm:516]: fixed in https://github.com/dankogai/p5-encode/commit/67f59dd4092c12043182e427dfaa5abd5c791e58

**Pull request merged**